vb-be/src/modules/room/room.controller.ts

```diff
@@ -21,39 +22,21 @@ export class RoomController {
 
     try {
       if (!roomObj) {
-        throw new ErrorHandler(HttpStatus.Error, ErrorText.InvalidDataSet);
+        throw new ErrorHandler(HttpStatus.Error, ErrorText.Forbidden);
       }
 
-      const tagsArr = [];
-      for (const tag of roomObj.tags) {
-        if (tag.__isNew__) {
-          const newTag = await this.tagService.create(tag.label);
-          tagsArr.push(newTag._id);
-        } else if (tag._id) {
-          const updatedTag = await this.tagService.updateScore(tag._id);
-          tagsArr.push(updatedTag._id);
-        } else {
-          throw new ErrorHandler(HttpStatus.Error, ErrorText.InvalidDataSet);
-        }
-      }
+      roomObj.tags.map(
+        async (tag) => await this.tagService.updateOrInsert(tag),
```

Comment:
I think we may need to `Promise.all()` this - the reason I had implemented it in a for-loop is because `.map` doesn't handle async how we might expect, this just returns an array of promises.

Reply:
it doesn't, i tried it out. its all the tag objects :O

Reply:
That's weird, everywhere I look has referenced it being a synchronous function, so it requires some sort of promise method

Reply:
I'll try to find some official documentation, everything has just been from stackoverflow and medium - I've encountered the need to do something like this before and could never get it to work otherwise